### ISSUE TITLE
Raster transparency for browsers without webp support.

### ISSUE THREAD

**Issue by sharkinsspatial** [mapbox/mapbox-gl-js#2263]
**mapbox-gl-js version**:
0.15.0
### Steps to Trigger Behavior

 1.  Add a raster source which contains nodata values which require an alpha channel to display as transparent.
 2. Attempt to view in browser which does not [support](http://caniuse.com/#feat=webp) webp

### Expected Behavior
No data values should display as transparent using an alpha channel.  See https://jsbin.com/nuyuyiveva/edit?html,js,output
<img width="1276" alt="screen shot 2016-03-14 at 1 42 18 pm" src="https://cloud.githubusercontent.com/assets/3671262/13752761/069bd310-e9ef-11e5-8c8e-311e36461a10.png">

### Actual Behavior
No data values are displayed as black when tile requests fall back to `.jpg` (no alpha channel) in browsers without `.webp` support.
<img width="1092" alt="screen shot 2016-03-14 at 1 43 05 pm" src="https://cloud.githubusercontent.com/assets/3671262/13752863/63c70230-e9ef-11e5-9a53-d8615ea056aa.png">

I can make requests for `.png` tiles from an endpoint, but the the urls in the tiles property of the TileJSON for an uploaded `.tiff` always have a `.jpg` extension.  Here is the [TileJSON](https://api.mapbox.com/v4/sharkins.13mpevkf.json?access_token=pk.eyJ1Ijoic2hhcmtpbnMiLCJhIjoiOFRmb0o1SSJ9.PMTjItrVcqcO8xBfgP1pMw) for the image above. I realize that I can force .png use with embeded tilejson properties directly as described [here](https://github.com/mapbox/mapbox-gl-js/issues/1616) but then I lose the retina checking and webp support check provided by [normalizeTileUrl](https://github.com/mapbox/mapbox-gl-js/blob/e752a88be1ad24545472a2ad93283741724e3962/js/util/mapbox.js#L68).

Currently [normalizeTileUrl](https://github.com/mapbox/mapbox-gl-js/blob/master/js/util/mapbox.js#L68) will use always use `.jpg` when there is no `.webp` support.  It seems like the two approaches to circumvent this are 

1. Modify the upload so that the tiles property in the resulting TileJSON uses the `.png` extension.

2. Extend [RasterTileSource](https://github.com/mapbox/mapbox-gl-js/blob/master/js/source/raster_tile_source.js) to include an alpha channel flag so that it can fall back gracefully to `.png` when there is no `.webp` support.

I am unsure if 1 is possible, and I'm not sure if 2 would be the right approach or not.

**Comment by lucaswoj** [comment]
Thank you very much for this descriptive and thoughtful ticket :bow:

There are several long-term solutions to your problem. We should consider perusing each of them:

 - allow TileJSON to indicate an endpoints "available extensions in tilejson in order of preference", as proposed in https://github.com/mapbox/mapbox-gl-js/issues/1725#issue-116862606
 - generalize format and retina detection and handling to vanilla GeoJSON sources through custom content negotiation headers
 - add a style property to raster layers to indicate that transparency is required and request `.png`s instead of `.jpg`s

Thoughts @ansis @jfirebaugh @mourner @bhousel?

**Comment by jfirebaugh** [comment]
I think this is a bug in the Mapbox TileJSON API endpoint. My understanding is it is supposed to detect the use of raster transparency, and use the `.png` extension in the `tiles` property when that's the case.

cc @yhahn

**Comment by jfirebaugh** [comment]
Previously: #1725.

**Comment by lucaswoj** [comment]
Sounds like this is an upstream "won't fix."

> @yhahn: mm, no we don't use png in these cases
> @yhahn: if the original upload was a raster we'll serve tilejson with jpg always now
> @yhahn: png tiles for this imagery is prohibitively expensive in terms of bandwidth

**Comment by lucaswoj** [comment]
@sharkinsspatial This is an upstream issue related to the Mapbox API. I'll my best escalate the issue internally.

**Comment by timiyay** [comment]
Is this issue now unblocked, with the recent release of `libwebp.js` v0.6.0?

To handle cross-browser raster tile transparency, I'm currently adding the external raster source at runtime. This suffices, but takes my client out of their Mapbox Studio workflow, and requires them to work with tile hosting solutions and config.

Could the workaround be made redundant by using `libwebp.js`?
